fedbiomed/transport/client.py

```diff
@@ -140,6 +220,31 @@ async def send(self, message: Message) -> None:
         await self._sender.send(message)
 
 
+    async def _connect(self):
+        """Updates connection state and dispatch event to run listeners
+
+        This method also implements auto-trust for server certificate
+        """
+
+        while True:
+            if is_server_alive(self._researcher.host, self._researcher.port):
+                # Gets server certificate before creating the channel
+                # This implementation assumes that the provided IP and PORT trusted
+                self._researcher.certificate = \
```

Comment:
Unless I'm mistaking, the server certificate is never saved by the node in this version, each time the node connects to the researcher it picks the current certificate for that session.

This is perfectly OK for this first version, but if you confirm this, I'd like to add an explicit comment for future developer that we have a hardcoded policy here, which is OK for honest but curious but still enables malicious MITM attacks.

Reply:
I added a few lines to your existing comment